compiler/rustc_trait_selection/src/solve/eval_ctxt.rs

```diff
@@ -156,17 +163,31 @@ impl<'tcx> InferCtxtEvalExt<'tcx> for InferCtxt<'tcx> {
             var_values: CanonicalVarValues::dummy(),
             nested_goals: NestedGoals::new(),
             tainted: Ok(()),
-            inspect: (self.tcx.sess.opts.unstable_opts.dump_solver_proof_tree
-                || matches!(generate_proof_tree, GenerateProofTree::Yes))
-            .then(ProofTreeBuilder::new_root)
-            .unwrap_or_else(ProofTreeBuilder::new_noop),
+            inspect: {
+                if self.tcx.sess.opts.unstable_opts.dump_solver_proof_tree
+                    == SolverProofTreeCondition::Always
+                {
+                    generate_proof_tree = GenerateProofTree::Yes(
+                        match self.tcx.sess.opts.unstable_opts.dump_solver_proof_tree_uses_cache {
+                            true => DisableGlobalCache::No,
+                            false => DisableGlobalCache::Yes,
+                        },
+                    )
+                }
+
+                match generate_proof_tree {
+                    GenerateProofTree::No => ProofTreeBuilder::new_noop(),
+                    GenerateProofTree::Yes(global_cache_disabled) => {
+                        ProofTreeBuilder::new_root(global_cache_disabled)
+                    }
+                }
+            },
         };
         let result = ecx.evaluate_goal(IsNormalizesToHack::No, goal);
 
         let tree = ecx.inspect.finalize();
         if let Some(tree) = &tree {
-            // module to allow more granular RUSTC_LOG filtering to just proof tree output
-            super::inspect::dump::print_tree(tree);
+            println!("{tree:?}");
```

Comment:
here ^^

Reply:
thx, because I forgot to change this to use stdout